### DISCUSSION TITLE
Receive message broadcasted from other application

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#3008] [Q&A]
Hi there,
I am using `@itwin/web-viewer-react` to create an app which is displayed within an iFrame in my main application. I want to send some message ( specifically _element id_) from main application's iframe to the one that contains this viewer app. This broadcasted _element id_ would then be used to perform some other actions in viewer app like zoom, hilighting the marker, display data in widget based on that element id, etc. I am trying `window.addEventListener` to catch the event and then fetch the element id from that but not sure whether i'm in the right direction...
What should be the right approach to achieve this? Is there any sample codes for this?

Thanks

**Comment by kckst8** [comment]
@kmkashyap Have you tried using messages? https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage

**Reply by kmurari103** [reply]
I tried using the message one and it worked for iframes communication but then i started facing issue- when i clicked on toolbar buttons (any from vertical or horizontal) in viewer app , that `evenhandler `gets called for these events as well. But since these button clicks does not broadcast intended message (element id), so it throws error.
